Dockerfile

```diff
@@ -1,33 +1,30 @@
 # Pull base image.
-# TODO: port to a non-centos based image
-FROM centos/python-38-centos7
+FROM rockylinux:8
 
 # Specifies which defines.conf file is copied into the container,
 # set as required via "docker build --build-arg VENUE=[ops|uat|sit|dev|test] ..."
 ARG VENUE=ops
 
 USER root
 
-# As of 2024, mirrorlist.centos.org doesn't exist anymore, so yum needs to be
-# repointed to the new endpoint (vault.centos.org) to pull packages from.
-# See https://serverfault.com/questions/1161816/mirrorlist-centos-org-no-longer-resolve/1161847#1161847
-# TODO: remove once ported off of centos
-RUN sed -i s/mirror.centos.org/vault.centos.org/g /etc/yum.repos.d/CentOS-*.repo \
-    && sed -i s/^#.*baseurl=http/baseurl=http/g /etc/yum.repos.d/CentOS-*.repo \
-    && sed -i s/^mirrorlist=http/#mirrorlist=http/g /etc/yum.repos.d/CentOS-*.repo
-
-# Install yum packages
-RUN yum -y update --security \
-    && yum -y install epel-release \
-    && yum -y install gcc yum-utils curl git tmux vim emacs-nox openssl-devel bzip2-devel libxml2-devel udunits2-devel libxslt-devel \
-    && yum -y install python3-mod_wsgi httpd-devel libaec-devel dnf net-tools apache2 netcdf-devel sqlite-devel hdf5-devel \
-    && yum -y install nbdkit-plugin-gzip perl-PerlIO-gzip hdf-devel \
-    && yum -y install logrotate cronie patch wkhtmltopdf Xvfb urw-fonts libXext \
-    && yum -y install udunits2-devel \
-    && yum -y install texlive \
-    && yum -y groupinstall "Development Tools" \
-    && yum -y install cairo-devel dejavu-sans-fonts \
-    && yum clean all
+# Install packages
+RUN dnf -y install 'dnf-command(config-manager)' && dnf config-manager --set-enabled powertools
+
+RUN dnf -y update --security \
+    && dnf -y install epel-release \
+    && dnf -y install python3.11 python3.11-pip python3.11-devel python3.11-numpy python3-mod_wsgi \
+    && dnf -y install gcc yum-utils git tmux vim emacs-nox net-tools httpd \
+    && dnf -y install openssl-devel bzip2-devel libxml2-devel udunits2-devel libxslt-devel \
+    && dnf -y install httpd-devel libaec-devel netcdf-devel sqlite-devel \
+    && dnf -y install hdf5 hdf5-devel hdf5-static nbdkit-gzip-filter perl-PerlIO-gzip libtirpc-devel \
+    && dnf -y install logrotate cronie patch Xvfb urw-fonts libXext \
+    && dnf -y install udunits2-devel \
+    && dnf -y groupinstall "Development Tools" \
+    && dnf -y install texlive cairo-devel dejavu-sans-fonts \
+    && dnf clean all
+
+# Install wkhtmltopdf
+RUN dnf -y localinstall https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6.1-2/wkhtmltox-0.12.6.1-2.almalinux8.x86_64.rpm
```

Comment:
Any particular reason why we want to break these dnf commands up into multiple RUN blocks/multiple dnf commands? Are each of the RUN blocks cacheable?

Reply:
Combined the `RUN` statements into a single statement. Merging the `dnf` commands is a little trickier, as I believe certain commands do need to be run prior to others (for example the `epel-release ` installation).